storage/oasis/nodeapi/file/kvstore.go

```diff
@@ -27,28 +29,95 @@ type KVStore interface {
 }
 
 type pogrebKVStore struct {
-	*pogreb.DB
+	db *pogreb.DB
 
 	path   string
 	logger *log.Logger
+
+	initialized uint32
 }
 
 var _ KVStore = (*pogrebKVStore)(nil)
 
+func (s *pogrebKVStore) isInitialized() bool {
+	return atomic.LoadUint32(&s.initialized) == 1
```

Comment:
Is there a scenario where using just plain value assignment and reading (instead of `atomic.*`) would get us into trouble?
If not, I'd slightly prefer avoiding `atomic` because it otherwise misleads the reader into thinking they're missing a complex scenario (which is sucky for refactors/edits later), and the complexity is unlikely to increase.

Reply:
We need an atomic (or a different form of synchronization, e.g. a lock) since the initialization is done in a separated go-routine. Without it, concurrent reads and writes can lead to undefined behavior. 

I can add a comment to the `initialzied` field to make it clear.